File: C++/Easy/121.买卖股票的最佳时机.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.size()<2) return 0;
        int buy=prices[0];
        int sell=0;
        int maxProfit=0;
        for(int i=1;i<prices.size();i++)
        {
            if(prices[i]-prices[i-1]>0)
            {
                if(prices[i]-buy>maxProfit) 
                {
                    sell=prices[i];
                    maxProfit=sell-buy;
                }
            }
            else
            {
                if(prices[i]<buy) buy=prices[i];
            }
        }
        return maxProfit;
    }
}; 

File: python/Easy/121. 买卖股票的最佳时机.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        curP = 0
        for i in range(1,len(prices)):
            temp = prices[i] - prices[i-1]
            if curP + temp > 0:
                curP = curP + temp
            else:
                curP = 0
            maxP = max(curP, maxP)
        return maxP
